client/adventureMap/AdventureMapInterface.cpp

```diff
@@ -957,3 +944,35 @@ void AdventureMapInterface::onScreenResize()
 	if (widgetActive)
 		activate();
 }
+
+bool AdventureMapInterface::isValidAdventureSpellTarget(int3 targetPosition, const CGObjectInstance * topObjectAtTarget, SpellID spellId)
```

Comment:
Again - why not move this to CSpell or to AdventureMapMechanics class (which is member of CSpell)?